src/plugins/Plugin.h

```diff
@@ -30,6 +30,9 @@ struct Plugin {
     virtual void OnUpgradeCompleted(sc2::UpgradeID) {
     }
 
+    virtual void OnUnitEnterVision(const sc2::Unit*, Builder*) {
```

Comment:
A Strategy (and other plugins) will want to build certain units right away, in response to seeing certain enemies. This will be directly used during scouting.
Such as, "See Dark Shrine / Cloaked ground units, build detection at base / Ravens".

Reply:
Is Builder* okay to leave in?

Reply:
Yes, this is totally fine.